frontend/src/metabase/visualizations/components/ChartHeader.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback } from "react";
+import PropTypes from "prop-types";
+import { iconPropTypes } from "metabase/components/Icon";
+import { ChartHeaderRoot } from "./ChartHeader.styled";
+
+const propTypes = {
+  series: PropTypes.array.isRequired,
+  settings: PropTypes.object.isRequired,
+  icon: PropTypes.shape(iconPropTypes),
+  actionButtons: PropTypes.node,
+  onChangeCardAndRun: PropTypes.func,
+};
+
+const ChartHeader = ({
```

Comment:
Ideally this should be a hook, but we are going to replace the component used in classes only, so I made it a component as well.